handlers.go

```diff
@@ -50,19 +82,40 @@ func (hh *httpHandlers) ping(w http.ResponseWriter, r *http.Request) {
 
 //goodToGo returns a 503 if the healthcheck fails - suitable for use from varnish to check availability of a node
 func (hh *httpHandlers) goodToGo(writer http.ResponseWriter, req *http.Request) {
-	if _, err := hh.checker(); err != nil {
+	if resp, err := hh.checkWriterAvalailability(); err != nil {
+		log.Info(resp)
 		writer.WriteHeader(http.StatusServiceUnavailable)
 	}
 
 }
 
+func (hh *httpHandlers) checkWriterAvalailability() (string, error) {
+	var endpointsToCheck []string
+	for _, baseURL := range hh.baseURLMap {
+		reg, _ := regexp.Compile("\\w*$")
+		//g2g := reg.ReplaceAllLiteralString(baseUrl, "__gtg")
+		endpointsToCheck = append(endpointsToCheck, reg.ReplaceAllLiteralString(baseURL, "__gtg"))
+	}
+	writer, goodToGo, gtgErr := checkWriterStatus(endpointsToCheck)
+	if goodToGo == false {
```

Comment:
Because its checking n number of writers I wanted a way to track if all checks were ok and this was what I came up with? Not sure if its correct or not because I cant get the goodToGo to return anything